### PR TITLE
Nataliia romanchuk/homework8

### PR DESCRIPTION
**Pull request by Vivanatalya**

### CODE REVIEW

File: nataliia_romanchuk/homework_8/every_person.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        self.birth_date = datetime.strptime(birth_date, "%d.%m.%Y")
+        self.job = job
+        self.working_years = working_years
+        self.salary = salary
+        self.country = country
+        self.city = city
+        self.gender = gender
+
+    def name(self):
+        return f"{self.first_name}{' '}{self.last_name}"
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework_8/every_person.py:21]: return f"{self.first_name} {self.last_name}"

File: nataliia_romanchuk/homework_8/every_person.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    def age(self):
+        now = datetime.now()
+        not_full_year = (now.month, now.day) < (self.birth_date.month, self.birth_date.day)
+        return now.year - self.birth_date.year - not_full_year
+
+    def work(self):
+        return f"{self.PREFIX[self.gender]} a {self.job}"
+
+    def money(self):
+        return f"{self.salary * self.working_years*12:_}".replace("_", " ")
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework_8/every_person.py:32]: missing spaces around oeprator *

File: nataliia_romanchuk/homework_8/warior.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    attack = 7
+
+
+def fight(unit_1: Warrior, unit_2: Warrior):
+    while unit_1.is_alive:
+        unit_1.perform_attack(unit_2)
+        if not unit_2.is_alive:
+            break
+        unit_2.perform_attack(unit_1)
+    return unit_1.is_alive
```
Comment (kudinov-fedor) [inline nataliia_romanchuk/homework_8/warior.py:29]: empty liine missing

File: nataliia_romanchuk/homework_8/warior.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class Warrior:
+    health = 50
+    attack = 5
+
+    @property
+    def is_alive(self):
+        if self.health > 0:
+            return True
+        else:
+            return False
```
Comment (kudinov-fedor) [inline nataliia_romanchuk/homework_8/warior.py:10]: return self.health > 0

File: tests/nataliia_romanchuk/homework8/test_person.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    assert p1.work() == "He is a welder", "Person's Job"
+    assert p1.money() == "648 000", "Person's Money"
+
+
+def test_new_person2():
+    p2 = Person("Hanna Rose", "May", "05.12.1995", "designer", 2.2, 2150, "Austria", "Vienna")
+    assert p2.name() == "Hanna Rose May", "Person's Name"
+    assert p2.age() == 28, "Person's Age"
+    assert p2.work() == "Is a designer", "Person's Job"
+    assert p2.home() == "Lives in Vienna, Austria", "Person's Home"
```
Comment (kudinov-fedor) [inline tests/nataliia_romanchuk/homework8/test_person.py:17]: She is case not covered

**Review by kudinov-fedor — APPROVED** [review]

File: nataliia_romanchuk/homework_8/every_person.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        self.salary = salary
+        self.country = country
+        self.city = city
+        self.gender = gender
+
+    def name(self):
+        return f"{self.first_name} {self.last_name}"
+
+    def age(self):
+        now = datetime.now()
```
Comment (kudinov-fedor) [inline nataliia_romanchuk/homework_8/every_person.py:24]: as per requirements there should be specific date for comparrison

**Review by kudinov-fedor — CHANGES_REQUESTED** [review]
change in Person today date is required

File: tests/nataliia_romanchuk/homework8/test_person.py (hunk 5 above)
Comment (kudinov-fedor) [inline tests/nataliia_romanchuk/homework8/test_person.py:17]: how will you test that age works properly ?

Comment (Vivanatalya) [reply tests/nataliia_romanchuk/homework8/test_person.py:19]: added

Comment (Vivanatalya) [reply tests/nataliia_romanchuk/homework8/test_person.py:19]: I'm testing the age in 16 line. Or what what do you mean?

File: nataliia_romanchuk/homework_8/every_person.py (hunk 6 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/every_person.py:24]: I didn't get what was the question about. Here I return years return now.year - self.birth_date.year - not_full_year

Comment (fkudinov-mn) [reply nataliia_romanchuk/homework_8/every_person.py:24]: based on requirements:  now is  2018-01-01
but it is ok if it is left as is

File: nataliia_romanchuk/homework_8/three_chefs.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class UkrainianCook(AbstractCook):
+    def __init__(self):
+        super().__init__('Pampushky', 'Uzvar')
+
+
+class ItalianCook(AbstractCook):
+    def __init__(self):
+        super().__init__('Pizza', 'Juice')
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework_8/three_chefs.py:32]: you could move these params on class level since they are immutable:

```
class AbstractCook:
    food = "Food"
    drink = "Drink"

class UkrainianCook(AbstractCook):
        food = "Pampushky"
        drink = "Uzvar"
```

File: tests/nataliia_romanchuk/homework8/test_person.py (hunk 5 above)
Comment (fkudinov-mn) [reply tests/nataliia_romanchuk/homework8/test_person.py:19]: if birthday is before expected 
if birthday equals expected
if birthday is after expected

etc.


testing of boundary values

**Review by fkudinov-mn — APPROVED** [review]

File: nataliia_romanchuk/homework_8/every_person.py (hunk 6 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/every_person.py:24]: I prefer to left as it is, due to the lack of time.

File: nataliia_romanchuk/homework_8/three_chefs.py (hunk 7 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/three_chefs.py]: done

File: nataliia_romanchuk/homework_8/warior.py (hunk 4 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/warior.py]: done

File: nataliia_romanchuk/homework_8/warior.py (hunk 3 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/warior.py]: done

File: nataliia_romanchuk/homework_8/every_person.py (hunk 2 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/every_person.py]: added

File: nataliia_romanchuk/homework_8/every_person.py (hunk 1 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework_8/every_person.py]: done
